start_esp/nginx-auto.conf.template

```diff
@@ -115,19 +115,48 @@ http {
 
 % if cors_preset == 'basic':
     # Begin basic CORS preset
-    if (${'$'}request_method = 'OPTIONS') {
-      add_header 'Access-Control-Max-Age' 1728000;
-      add_header 'Access-Control-Allow-Origin' '${cors_allow_origin}';
-      add_header 'Access-Control-Allow-Methods' '${cors_allow_methods}';
-      add_header 'Access-Control-Allow-Headers' '${cors_allow_headers}';
-      add_header 'Content-Type' 'text/plain; charset=utf-8';
-      add_header 'Content-Length' 0;
-      return 204;
+  % if cors_allow_origin_regex != '':
+    if ($http_origin ~* (${cors_allow_origin_regex})) {
+        set $cors "true";
+    }
+  % else:
+    set $cors "true";
+  % endif
+
+    if ($request_method = 'OPTIONS') {
+        set $cors "${'$'}{cors}options";
+    }
+
+    if ($cors = "trueoptions") {
+        add_header 'Access-Control-Max-Age' 1728000;
+      % if cors_allow_origin_regex != '':
+        add_header 'Access-Control-Allow-Origin' "$http_origin";
+      % else:
+        add_header 'Access-Control-Allow-Origin' '${cors_allow_origin}';
```

Comment:
can we define a new variable $allow_origin for this?  it can be set in line 118, so we can avoid another "if" check

Reply:
I see. I'll improve it.